cmd/utils/diskusage.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// +build !windows
+
+package utils
+
+import (
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/sys/unix"
+)
+
+var stat unix.Statfs_t
+
+func getFreeDiskSpace(path string) uint64 {
+
+	err := unix.Statfs(path, &stat)
+	if err != nil {
+		log.Warn("Failed to call Statfs", "path", path, "err", err)
+		sigtermCh <- syscall.SIGTERM
+		return 0
```

Comment:
I don't think this method should send a sigterm on some package-global channel, instead it should just return `(uint64, error)`. The caller can decide whether to quit or not, but the utility-helper doesn't need to be aware of the caller environment.

Reply:
So basically
```
return 0, fmt.Errorf("Failed to call Statfs: %v", err)
```

Reply:
Also, if we're on some platform where we can't find a suitable system-call to obtain this info, it will always return `0, error`. And I think we should just exit the monitoring-loop with a suitable log message, but not send the sigkill. One should still be able to run geth, even if the disk-check isn't working properly

Reply:
all of the above comments have been implemented